### DISCUSSION TITLE
Wrong mesh dimension -1 error while 2D meshing: 3D trapezium created by "hspace" functionality and a BUG ISSUE IN TESSELATION (with reg 1)

### DISCUSSION THREAD

**Discussion by SrijanY** [neperfepx/neper#862] [Q&A] [answered]
I made a 3D Trapezium, tesselation is done. But while meshing, it stops with the error.
 Gmsh is installed perfectly with all tests run and I've been using it for a long time it works fine.

I have attached the -T command, -M command and the image of the tesselation generated.

-T :
![Tesselation](https://github.com/neperfepx/neper/assets/142650311/6fc55c33-f893-48e1-a78d-89fed180da49)

-M :
[
![Meshing](https://github.com/neperfepx/neper/assets/142650311/46718783-f73e-457b-99eb-461226b045e6)

Generated Tesselation:
![frus12](https://github.com/neperfepx/neper/assets/142650311/8bb47851-fd2e-4d2b-a095-55250308d073)
Please Help!!
Thankyou 

I also tried by putting regularisation, but it gave a bug issue.
Here's that command:
![Screenshot (192)](https://github.com/neperfepx/neper/assets/142650311/0c34a7f5-15a2-4240-bdbe-f05ef2830b92)

I tried visualising the mesh for some different elements other than tet and triangle, this showed some missing portion in the trapezium shape.
Here's the image:

![Screenshot (194)](https://github.com/neperfepx/neper/assets/142650311/53627e64-4aa7-4bbe-b963-018879377d2e)

**Comment by Srijan7Y** [comment, marked as answer]
Issue is solved. I took the left corner of the trapezium not at origin, but slighlty ahead like from 0.5 , and increased my domain to 2.0 for the same. Thus by avoiding the shape corner from lying on origin, this issue was solved. 
This is the updated command, which didn't cause any issues,

**$ neper -T -n 100 -morpho gg -morphooptistop "val=1e-2" -domain "cube(2.5,0.5,1.2)" -transform "cut(hspace(2.4,1.2,0,0.5),hspacei(0.6,1.2,0,-0.5))"** 

The shape got easily meshed after this.
